ui/common/src/dialog.ts

```diff
@@ -159,56 +195,49 @@ class DialogWrapper implements Dialog {
     if (dialogPolyfill) dialogPolyfill.registerDialog(dialog); // ios < 15.4
 
     const justThen = Date.now();
-    const cancelOnInterval = () => Date.now() - justThen > 200 && this.close('cancel');
-
+    const cancelOnInterval = (e: PointerEvent) => {
+      if (Date.now() - justThen < 200) return;
+      const r = dialog.getBoundingClientRect();
+      if (e.clientX < r.left || e.clientX > r.right || e.clientY < r.top || e.clientY > r.bottom)
```

Comment:
that might conflict with (user) tooltips, select, mselect and anything adding a layer on top of the window.

Reply:
assuming they preventDefault on the click, it shouldn't arrive on the body element. If it does, I think we still want to close in that case.

I added the bounding client rect check because non-modal dialogs don't trap background clicks for us like modal ones. That said, I'm not super attached to even allowing/supporting click away for non-modals.